src/Mjolnir.ts

```diff
@@ -952,4 +954,17 @@ export class Mjolnir {
     public async processRedactionQueue(roomId?: string): Promise<RoomUpdateError[]> {
         return await this.eventRedactionQueue.process(this.client, roomId);
     }
+
+    public async handleServerAbuseReport(roomId: string, reporterId: string, event: any, reason: string): Promise<any> {
+        this.reportManager.handleServerAbuseReport(roomId, reporterId, event, reason);
+
+        if (Object.keys(this.protectedRooms).includes(roomId)) {
+            if (event['sender'] === await this.client.getUserId()) return; // Ignore ourselves
+
+            // Iterate all the enabled protections
+            for (const protection of this.getEnabledProtections()) {
+                await protection.handleReport(this, roomId, reporterId, event, reason);
```

Comment:
So every single protection can accept reports? Maybe but that feels... odd.

Reply:
can't really think of a more generic way to do it. would be strange to have one single nominated report-handling protection

Reply:
Couldn't we use an `EventEmitter` to let individual protections register and get the information they need?

Reply:
we don't currently have a neat way to tell protections they're being unregistered so they can clean up subscriptions

Reply:
Can we unregister a protection at all?

Reply:
https://github.com/matrix-org/mjolnir/blob/ff9a7db159b95c1c1e196a022908e4bde1d6a081/src/Mjolnir.ts#L529
and integration tests may want to do it

Reply:
Ok, fair enough.
Can you file a followup issue, though?

Reply:
https://github.com/matrix-org/mjolnir/issues/215